drt_1d.py

```diff
@@ -32,47 +34,152 @@ def construct_source_term(num_voxels:int, source_voxel_index:int, source_strengt
     logging.debug("Constructed source vector: %s", s)
     return s
 
-def calculate_K_trans(ri:int, rj:int, voxel_size:float, sigma_s:np.ndarray):
+def calculate_K_trans(start_index:int, end_index:int, ri:float, rj:float, voxel_size:float, sigma_s:np.ndarray):
     """
     Calculate the streaming operator. Used by the construct_transition_matrix function to build the K_trans matrix.
 
-    Current method calculates tau by summing the cross-sections of the voxels from the start index to the end index, 
-    always summing from the left edge of the starting voxel to the left edge of the ending voxel, regardless of the direction of transport.
-
     Parameters:
-    - ri (int): Index of the starting voxel.
-    - rj (int): Index of the ending voxel.
-    - voxel_size (float): Size of each voxel.
+    - start_index (int): Starting position voxel index
+    - end_index (int): Ending position voxel index
```

Comment:
You should be able to compute/find `start_index` and `end_index` from `ri` and `rj` and the voxel grid.  This will prevent user error.

This is particularly easy for a fixed voxel size, but you may want to plan on dong it for an arbitrary voxel size.

Reply:
Possible solution for arbitrary voxel size: user inputs dictionary similar to
`voxels: {
            "index": 0,
            "start_position": 0.0,
            "end_position": 1.0,
            "cross_section": 0.2,
            "positions": [0.0, 0.1, 0.4]
        },
        {...}`